### PR TITLE
[Admin][Dashboard] only fulfilled state orders will be counted in statistics

### LINKED ISSUE #8251: Statistics in show customer in admin dashboard, count also orders in state awaiting_payment
Hello.

One customer with order to bank transfer, has a order with following state.
order-mysql

I understand that in statistics of user, this order should not be included.

statistics_user

### PR DESCRIPTION
**Pull request by mertoksuz**
| Q               | A
| --------------- | ---
| Bug fix?        | yes |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets | fixes #8251 |
| License         | MIT |

### CODE REVIEW

**Comment by Zales0123** [comment]
There are some failing behat scenarios. Could you please look at them and fix?

**Comment by mertoksuz** [comment]
updated now, thanks

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php [hunk 1]
```diff
@@ -218,8 +220,10 @@ public function countByChannel(ChannelInterface $channel)
             ->select('COUNT(o.id)')
             ->andWhere('o.channel = :channel')
             ->andWhere('o.state NOT IN (:states)')
+	        ->andWhere('o.paymentState NOT IN (:payment_states)')
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php]: Wrong indent

Comment (mertoksuz) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php]: updated, thanks

**Comment by mertoksuz** [comment]
@pamil you are right :)

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php [hunk 2]
```diff
@@ -203,9 +203,9 @@ public function getTotalSalesForChannel(ChannelInterface $channel)
         return (int) $this->createQueryBuilder('o')
             ->select('SUM(o.total)')
             ->andWhere('o.channel = :channel')
-            ->andWhere('o.state NOT IN (:states)')
+            ->andWhere('o.state IN (:states)')
```
Comment (pamil) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php]: `o.state = :state`?

File: src/Sylius/Component/Core/Repository/OrderRepositoryInterface.php [hunk 3]
```diff
@@ -106,7 +106,7 @@ public function getTotalSalesForChannel(ChannelInterface $channel);
      *
      * @return int
      */
-    public function countByChannel(ChannelInterface $channel);
+    public function countFulfilledByChannel(ChannelInterface $channel);
```
Comment (pamil) [inline src/Sylius/Component/Core/Repository/OrderRepositoryInterface.php]: Could you add a note to `UPGRADE.md`? :)

File: src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php (hunk 2 above)
Comment (mertoksuz) [inline src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php]: yes, updated. thank you.

File: src/Sylius/Component/Core/Repository/OrderRepositoryInterface.php (hunk 3 above)
Comment (mertoksuz) [inline src/Sylius/Component/Core/Repository/OrderRepositoryInterface.php]: note added :+1:

**Comment by mertoksuz** [comment]
@pamil please review :)

File: UPGRADE.md [hunk 4]
```diff
@@ -7,6 +7,9 @@
 * In order to be compatibile with Doctrine ORM 2.6+ and be more consistent 
   `OrderRepositoryInterface::count()` signature was changed to `OrderRepositoryInterface::countPlacedOrders()`.
 
+* In statistics to be correct data with orders informations
```
Comment (mheki) [inline UPGRADE.md]: How about `In statistics to display information about fulfilled orders` instead?

File: features/admin/dashboard.feature [hunk 5]
```diff
@@ -17,22 +17,22 @@ Feature: Statistics dashboard in a single channel
         Given 3 customers have placed 4 orders for total of "$8566.00"
         And then 2 more customers have placed 2 orders for total of "$459.00"
         When I open administration dashboard
-        Then I should see 6 new orders
+        Then I should see 0 new orders
         And I should see 5 new customers
-        And there should be total sales of "$9,025.00"
-        And the average order value should be "$1,504.17"
+        And there should be total sales of "$0.00"
+        And the average order value should be "$0.00"
```
Comment (pamil) [inline features/admin/dashboard.feature]: We should make the orders included in the statistics, seeing zeros everywhere isn't very useful :)

File: UPGRADE.md (hunk 4 above)
Comment (mertoksuz) [inline UPGRADE.md]: will be better, thanks.

**Comment by pjedrzejewski** [comment]
@mertoksuz Hi Mert! It's been a while, but are you willing to finish this PR? :)

**Comment by pjedrzejewski** [comment]
Fixed in #8520, thank you @mertoksuz for reporting & your work but we could not wait for the answer. :)

**Pull request closed without merging**